Sources/JS/initializeTypeCollection.js

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+function Type() {}
+Type.prototype.serialize = function() {
+    var result = {}
+    if(this.possibleType != null) result.possibleType = this.possibleType
+    if(this.definiteType != null) result.definiteType = this.definiteType
+    if(this.group != null) result.group = this.group
+    if(this.properties != null) result.properties = this.properties.slice()
+    if(this.methods != null) result.properties = this.methods.slice()
+    if(this.signature != null) result.signature = this.signature.serialize()
+
+    return result
+}
+
+function FunctionSignature() {}
+FunctionSignature.prototype.serialize = function() {
+    var result = {}
+    if(this.inputTypes != null) {
+        result.inputTypes = []
+        for (const inputType in this.inputTypes) {
+            result.inputTypes.append(inputType.serialize())
+        }
+    }
+    if(this.outputType != null)  result.outputType = this.outputType.serialize()
+
+    return result
+}
+
+var types = {}
```

Comment:
Maybe make a comment here like `// Must be kept in sync with TypeSystem.swift` to explain where these constants come from?